src/astgen

```diff
@@ -687,6 +816,24 @@ for typen in sorted(Classes.keys()):
                 "%Error: Non-final AstNode subclasses must be named AstNode*: Ast"
                 + typen)
 
+for typen in Classes.keys():
+    if "Node" in typen:
+        continue
+    bases = subclasses_of(typen)
+    if "NodeMath" not in bases:
+        continue
+    elif "NodeUniop" in bases:
+        arity = 1
+    elif "NodeBiop" in bases:
+        arity = 2
+    elif "NodeTriop" in bases:
+        arity = 3
+    elif "NodeQuadop" in bases:
+        arity = 4
```

Comment:
I will think about this, but realistically these are so basic we never change them. I agree, in general it would be nice to have the expression trees be more declarative.